receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -64,31 +65,20 @@ func (s *scraper) scrape(_ context.Context) (pdata.Metrics, error) {
 		return md, scrapererror.NewPartialScrapeError(err, metricsLen)
 	}
 
-	initializeCPUTimeMetric(metrics.AppendEmpty(), s.startTime, now, cpuTimes)
-	return md, nil
-}
-
-func initializeCPUTimeMetric(metric pdata.Metric, startTime, now pdata.Timestamp, cpuTimes []cpu.TimesStat) {
-	metadata.Metrics.SystemCPUTime.Init(metric)
-
-	ddps := metric.Sum().DataPoints()
-	ddps.EnsureCapacity(len(cpuTimes) * cpuStatesLen)
+	s.mb.EnsureDataPointsCapacity(len(cpuTimes) * cpuStatesLen)
```

Comment:
No need correct? We can approximate this as len(States) * runtime.NumCPUs() and pass during New call.

Reply:
Yes, the usage usage can be removed here, but I'm still not sure if we should remove the method https://github.com/dmitryax/opentelemetry-collector-contrib/pull/1#discussion_r742311317